frontend/test/metabase/scenarios/embedding/embedding-native.cy.spec.js

```diff
@@ -0,0 +1,212 @@
+        "display-name": "Order ID",
+        type: "dimension",
+        dimension: ["field", ORDERS.ID, null],
+        "widget-type": "id",
+        default: null,
+      },
+      created_at: {
+        id: "a21ca6d2-f742-a94a-da71-75adf379069c",
+        name: "created_at",
+        "display-name": "Created At",
+        type: "dimension",
+        dimension: ["field", ORDERS.CREATED_AT, null],
+        "widget-type": "date/quarter-year",
+        default: null,
+      },
+      total: {
+        id: "68350949-02cc-f540-86cf-ddcda07529d8",
+        name: "total",
+        "display-name": "Total",
+        type: "dimension",
+        dimension: ["field", ORDERS.TOTAL, null],
+        "widget-type": "number/>=",
+        default: [0],
+        required: true,
+      },
+      source: {
+        id: "44038e73-f909-1bed-0974-2a42ce8979e8",
+        name: "source",
+        "display-name": "Source",
+        type: "text",
+      },
+      state: {
+        id: "88057a9e-91bd-4b2e-9327-afd92c259dc8",
+        name: "state",
+        "display-name": "State",
+        type: "dimension",
+        dimension: ["field", PEOPLE.STATE, null],
+        "widget-type": "string/!=",
+        default: null,
+      },
+      productId: {
+        id: "c967d72e-3687-aa01-8c47-458f7905305f",
+        name: "productId",
+        "display-name": "Product ID",
+        type: "number",
+        default: null,
+      },
+    },
+    query,
+    type: "native",
+  },
+};
+
+describe("scenarios > embedding > native questions", () => {
+  beforeEach(() => {
+    cy.intercept("GET", "/api/session/properties").as("sessionProperties");
+    cy.intercept("PUT", "/api/card/*").as("publishChanges");
+
+    restore();
+    cy.signInAsAdmin();
+
+    // Remap Product ID -> Product Title
+    cy.request("POST", `/api/field/${ORDERS.PRODUCT_ID}/dimension`, {
+      name: "Product ID as Title",
+      type: "external",
+      human_readable_field_id: PRODUCTS.TITLE,
+    });
+
+    cy.createNativeQuestion(questionDetails, { visitQuestion: true });
+
+    cy.icon("share").click();
+    cy.findByText("Embed this question in an application").click();
+    cy.wait("@sessionProperties");
+  });
+
+  it("should not display disabled parameters", () => {
+    cy.button("Publish").click();
+    cy.wait(["@publishChanges", "@publishChanges"]);
+
+    cy.document().then(doc => {
+      const iframe = doc.querySelector("iframe");
+
+      cy.signOut();
+      cy.visit(iframe.src);
+    });
+
+    cy.contains("Lora Cronin");
+    cy.contains("Organic");
+    cy.contains("39.58");
+
+    filterWidget().should("not.exist");
+  });
+
+  it("should display and work with enabled parameters while hiding the locked one", () => {
+    setParameter("Order ID", "Editable");
+    setParameter("Created At", "Editable");
+    setParameter("Total", "Locked");
+    setParameter("State", "Editable");
+    setParameter("Product ID", "Editable");
+
+    // We must enter a value for a locked parameter
```

Comment:
Isn't this strange given the fact that we already set the default value for this filter?

Reply:
No, expected. When you set to "Locked", then it basically means that the embedding will be defining the value and it has to be defined, it's non-optional.